### PR TITLE
APIMF-2867: final amf configuration interface

### PR DESCRIPTION
**Pull request by pope1838**

### CODE REVIEW

File: shared/src/main/scala/amf/client/remod/AMFClient.scala [hunk 1]
```diff
@@ -11,29 +11,30 @@ import amf.core.resolution.pipelines.ResolutionPipeline
...
+import scala.concurrent.{ExecutionContext, Future}
 
 // client
-private[remod] class AMFClient(env: BaseEnvironment) {
+private[amf] class AMFClient(val configuration: AMFConfiguration) {
 
+  implicit val exec: ExecutionContext = configuration.getExecutionContext
 
-  def getEnvironment:BaseEnvironment = env
+  def getEnvironment: AMFConfiguration = configuration
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFClient.scala]: If `configuration` is a public `val` then there is no need for this method. I would probably make the val private.

File: shared/src/main/scala/amf/client/remod/AMFClient.scala [hunk 2]
```diff
@@ -11,29 +11,30 @@ import amf.core.resolution.pipelines.ResolutionPipeline
...
 
-  def resolve(bu: BaseUnit): AmfResult = AMFResolver.resolve(bu, env) // clone? BaseUnit.resolved
+  def resolve(bu: BaseUnit): AmfResult = AMFResolver.resolve(bu, configuration) // clone? BaseUnit.resolved
 
-  def render(bu:BaseUnit, target:Vendor):String = AMFRender.render(bu, env)
+  def render(bu: BaseUnit, target: Vendor): String = AMFRender.render(bu, configuration)
   // bu.clone?
-  def validate(bu: BaseUnit, profileName: ProfileName): AMFValidationReport = ??? // how we can handle the parsing validations? error handler at base unit?
+  def validate(bu: BaseUnit, profileName: ProfileName): AMFValidationReport =
+    ??? // how we can handle the parsing validations? error handler at base unit?
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFClient.scala]: [Discussion] - We should decouple the ErrorHandler from BaseUnit as they are separate concepts. A BaseUnit for itself represent the information structure. Conceptually, it shouldn't know what is an `error`. Errors are part of the process or are derived from the unit structure, are not inheritly (?) part of it.

File: shared/src/main/scala/amf/client/remod/AMFConfiguration.scala [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    copy(resolvers = resolvers.withResourceLoaders(rl))
+
+  def withUnitCache(cache: UnitCache): AMFConfiguration =
+    copy(resolvers.withCache(cache))
+
+  def withPlugin(amfPlugin: AMFParsePlugin): AMFConfiguration = copy(registry = registry.withPlugin(amfPlugin))
+
+  def withPlugins(plugins: List[AMFPlugin[_]]): AMFConfiguration = copy(registry = registry.withPlugins(plugins))
+
+  // TODO: delete after branch?
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFConfiguration.scala]: If this will be deleted we should have it protected in the meantime

File: shared/src/main/scala/amf/client/remod/AMFConfiguration.scala [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  def removePlugin(id: String): AMFConfiguration = copy(registry = registry.removePlugin(id))
+
+  def withConstraints(profile: ValidationProfile): AMFConfiguration =
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFConfiguration.scala]: Troublesome that the method says `constraints` but the param is a `ValidationProfile` while the rest of the methods match this relationship.

File: shared/src/main/scala/amf/client/remod/AMFConfiguration.scala [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    copy(registry = registry.withConstraints(profile))
+
+  /**
+    * Merges two environments taking into account specific attributes that can be merged.
+    * This is currently limited to: registry plugins.
+    */
+  def merge(other: AMFConfiguration): AMFConfiguration = {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFConfiguration.scala:56]: Add a TODO so we don't forget.

File: shared/src/main/scala/amf/client/remod/AMFConfiguration.scala [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+import amf.client.remod.amfcore.registry.AMFRegistry
+import amf.core.validation.core.ValidationProfile
+import amf.internal.environment.Environment
+import amf.internal.reference.UnitCache
+import amf.internal.resource.ResourceLoader
+
+import scala.concurrent.ExecutionContext
+// all constructors only visible from amf. Users should always use builders or defaults
+
+class AMFConfiguration(private[amf] val resolvers: AMFResolvers,
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFConfiguration.scala]: Make it protected in the meantime

File: shared/src/main/scala/amf/client/remod/AMFConfiguration.scala [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+                       private[amf] val errorHandlerProvider: ErrorHandlerProvider,
+                       private[amf] val registry: AMFRegistry,
+                       private[amf] val logger: AMFLogger,
+                       private[amf] val listeners: List[AMFEventListener],
+                       private[amf] val options: AMFOptions) { // break platform into more specific classes?
+
+  def createClient(): AMFClient = new AMFClient(this)
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFConfiguration.scala:23]: General: use Seq or Set over List.

File: shared/src/main/scala/amf/client/remod/AMFConfiguration.scala [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  private[amf] def getParsingOptions: ParsingOptions        = options.parsingOptions
+  private[amf] def getRegistry: AMFRegistry                 = registry
+  private[amf] def getResourceLoaders: List[ResourceLoader] = resolvers.resourceLoaders
+  private[amf] def getUnitsCache: Option[UnitCache]         = resolvers.unitCache
+  private[amf] def getExecutionContext: ExecutionContext    = resolvers.executionContext.executionContext
+}
+
+object AMFConfiguration {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFConfiguration.scala]: Make protected in the meantime

File: shared/src/main/scala/amf/client/remod/AMFParser.scala [hunk 9]
```diff
@@ -16,33 +16,31 @@ import amf.internal.resource.{ResourceLoader, StringResourceLoader}
...
+  def parse(url: String, vendor: Vendor, env: AMFConfiguration): Future[AmfResult] = parseAsync(url, Some(vendor), env)
 
   /**
     * Asynchronously generate a BaseUnit from a given string.
     * @param stream: The unit as a string
     * @return A future that will have a BaseUnit or an error to handle the result of such invocation.
     */
   @JSExport
-  def parseStream(stream: String, env:AMFEnvironment): Future[AmfResult] = ???
+  def parseStream(stream: String, env: AMFConfiguration): Future[AmfResult] = ???
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFParser.scala]: Mmmh doubting about this. To me a `parseStream` method respects the premise that the result is also a Stream. What if I give AMF a never-ending Stream? Consuming the stream should be part of the client responsibility.

File: shared/src/main/scala/amf/client/remod/AMFParser.scala [hunk 10]
```diff
@@ -51,7 +49,7 @@ private[remod] object AMFParser {
     * @return A future that will have a BaseUnit or an error to handle the result of such invocation.
     */
   @JSExport
-  def parseStream(stream: String,vendor:Vendor, env:AMFEnvironment): Future[AmfResult] = {
+  def parseStream(stream: String, vendor: Vendor, env: AMFConfiguration): Future[AmfResult] = {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFParser.scala]: Same as comment above

File: shared/src/main/scala/amf/client/remod/AMFRender.scala [hunk 11]
```diff
@@ -5,8 +5,8 @@ import org.yaml.model.YDocument
 
 private[remod] object AMFRender {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFRender.scala]: `AMFRenderer`

File: shared/src/main/scala/amf/client/remod/AMFResolver.scala [hunk 12]
```diff
@@ -5,8 +5,8 @@ import amf.core.model.document.BaseUnit
 
 private[remod] object AMFResolver {
 
-  def resolve(bu:BaseUnit, env: BaseEnvironment):AmfResult = ???
+  def resolve(bu:BaseUnit, env: AMFConfiguration):AmfResult = ???
 
-  def resolve(bu:BaseUnit, pipeline: AMFResolutionPipeline, env: BaseEnvironment):AmfResult = ???
+  def resolve(bu:BaseUnit, pipeline: AMFResolutionPipeline, env: AMFConfiguration):AmfResult = ???
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFResolver.scala]: - In which cases would I want to resolve with a pipeline that is not registered in the Configuration? 
- Why wouldn't I just run the pipeline myself?
- How do I choose the pipeline I want to run (or how is it chosen) ?

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala [hunk 13]
```diff
@@ -2,6 +2,6 @@ package amf.client.remod.amfcore.config
 
 import amf.core.model.document.BaseUnit
 
-private[remod] trait AMFEventListener {
+private[amf] trait AMFEventListener {
   def event(eventKind: String, bu: BaseUnit)
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala]: We can take advantage of Scala's pattern matching to provide an `AmfEvent` instead of a simple string. The `AmfEvent` can hold much more information and could be usefull for the user (for example I imagine an approximated progress of the amount of document's parsed. That is not possible with just a plain string or at least will be limited to the formatting we give it.)

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFLogger.scala [hunk 14]
```diff
@@ -1,6 +1,6 @@
 package amf.client.remod.amfcore.config
 
-private[remod] class AMFLogger {
+private[amf] class AMFLogger {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/amfcore/config/AMFLogger.scala:4]: I would probably add a TODO in all the classes that will be public in the future so that we can clearly see it when reviewing.

File: shared/src/main/scala/amf/core/AMFSerializer.scala [hunk 15]
```diff
@@ -141,46 +143,51 @@ class AMFSerializer(unit: BaseUnit, mediaType: String, vendor: String, env: Rend
...
-                              file: String,
-                              unit: BaseUnit,
-                              mediaType: String,
-                              vendor: String,
-                              options: RenderOptions,
-                              shapeOptions: ShapeRenderOptions): Future[Unit] = {
-        new AMFSerializer(unit, mediaType, vendor, options, shapeOptions).renderToFile(platform, file)
-      }
-    })
+    RuntimeSerializer.register(
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/AMFSerializer.scala:166]: Why is this change needed?

File: shared/src/main/scala/amf/core/services/RuntimeCompiler.scala [hunk 16]
```diff
@@ -37,38 +37,43 @@ object RuntimeCompiler {
...
       case _ => throw new Exception("No registered runtime compiler")
     }
   }
 
   // could not add new environment in this method as it forces breaking changes in ReferenceHandler
   def forContext(compilerContext: CompilerContext,
-            mediaType: Option[String],
-            vendor: Option[String],
-            referenceKind: ReferenceKind = UnspecifiedReference)(implicit executionContext: ExecutionContext): Future[BaseUnit] = {
+                 mediaType: Option[String],
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/services/RuntimeCompiler.scala:65]: Formatting changes

File: shared/src/main/scala/amf/client/remod/AMFClient.scala (hunk 2 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFClient.scala]: Totally agree. That comment is old. You 've handled the problem that chased that comment creating a validation plugin to collect parsing validations.

File: shared/src/main/scala/amf/client/remod/AMFConfiguration.scala (hunk 3 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFConfiguration.scala]: I'm doing amf package private the class in the mean time. Also, I'm adding the TODOs

File: shared/src/main/scala/amf/client/remod/AMFConfiguration.scala (hunk 5 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFConfiguration.scala:56]: Are you talking about the merge logic and define which configurations will be merged?
I will create a story at the new modularization epic to tackle that.

File: shared/src/main/scala/amf/client/remod/AMFResolver.scala (hunk 12 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFResolver.scala]: This is going to be changed by @AgustinBettati . Today we talk about that.
Resolve method is going to recieve a BaseUnit, the configuration and a PipelineName

Comment (tomsfernandez) [reply shared/src/main/scala/amf/client/remod/AMFResolver.scala]: Why not change it now?

File: shared/src/main/scala/amf/core/AMFSerializer.scala (hunk 15 above)
Comment (pope1838) [reply shared/src/main/scala/amf/core/AMFSerializer.scala:166]: Formatting change

File: shared/src/main/scala/amf/client/remod/AMFResolver.scala (hunk 12 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFResolver.scala]: He is already working on that, and some analysis to check the arguments is being made.
On the other hand, the class is private for now, so no risk to have it merged.

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFLogger.scala (hunk 14 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/amfcore/config/AMFLogger.scala:4]: I'm adding "//TODO: ARM" as prefix on comments to know that something there has to be deleted.

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala [hunk 17]
```diff
@@ -3,5 +3,11 @@ package amf.client.remod.amfcore.config
 import amf.core.model.document.BaseUnit
 
 private[amf] trait AMFEventListener {
-  def event(eventKind: String, bu: BaseUnit)
+  def event(eventKind: EventKind, bu: BaseUnit)
 }
+
+sealed case class EventKind(name: String)
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala:9]: Should be a trait

File: shared/src/test/scala/amf/core/parser/DuplicateJsonKeysTest.scala [hunk 18]
```diff
@@ -12,20 +12,20 @@ import org.scalatest.{AsyncFunSuite, Matchers}
...
   override implicit def executionContext: ExecutionContext = ExecutionContext.Implicits.global
 
   test("Parsed JSON with duplicate keys has several warnings") {
     Core.init().asFuture.flatMap { _ =>
       val errorHandler = DefaultParserErrorHandler.withRun()
-      val url = "file://shared/src/test/resources/parser/duplicate-key.json"
-      val env = AMFConfiguration.predefined()
-      RuntimeCompiler(url, None, None, base = Context(platform), cache = Cache(), env, errorHandler = errorHandler).map {
+      val url          = "file://shared/src/test/resources/parser/duplicate-key.json"
+      RuntimeCompiler(url, None, None, base = Context(platform), cache = Cache(), errorHandler = errorHandler).map {
```
Comment (tomsfernandez) [inline shared/src/test/scala/amf/core/parser/DuplicateJsonKeysTest.scala:23]: Why did you remove the predefined configuration here?

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala (hunk 17 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala:9]: No

File: shared/src/test/scala/amf/core/parser/DuplicateJsonKeysTest.scala (hunk 18 above)
Comment (pope1838) [reply shared/src/test/scala/amf/core/parser/DuplicateJsonKeysTest.scala:23]: It wasn't being use. Runtime compiler gets the static configuration internally.

File: shared/src/main/scala/amf/client/remod/AMFParser.scala (hunk 9 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFParser.scala]: Are you saying that a parseStream method should return a stream? that not makes so much sense in our context..
If the stream never ends, then is client fault. The same premise applies to a resource loader, a cache, o even a plugged parser.  Transformation pipelines could throw exceptions and that is another problem. If the user wants to break something, then I think there is no su much we could do, except for ban the possibility to inject logic in the framework but that is just what we want to have.

The only weird thing that I notice is that we are receiving and String not a real Stream, so all this discussion doesn't have sense. Stream logic could be handled using resource loaders and the method should be call "parseInMemory" "parseChain" "parseText" or something like that.
But as is a different discussion maybe the best approach is to ignored in here and create something focused on that.

File: shared/src/main/scala/amf/client/remod/AMFRender.scala (hunk 11 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFRender.scala]: I had to google this one

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala (hunk 17 above)
Comment (tomsfernandez) [reply shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala:9]: why not? Using a case class as parent class prohibits me of using a case class subtype of `EventKind` which limits my event listening capabilities as a user

Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala:9]: Is basically an enum, not a model of behaviour or an abstraction. Using a sealed case class and objects, you close the set of possible values, but having the benefits of case class like unapply and equals.

Comment (tomsfernandez) [reply shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala:9]: why limit this behaviour? An Event can have a lot more information than just an enum.

**Review by tomsfernandez — DISMISSED** [review]

File: shared/src/main/scala/amf/client/remod/AMFRender.scala [hunk 19]
```diff
@@ -5,8 +5,8 @@ import org.yaml.model.YDocument
...
 
-  def render(bu:BaseUnit, env:BaseEnvironment): String = ???
+  def render(bu:BaseUnit, env: AMFConfiguration): String = ???
```
Comment (AgustinBettati) [inline shared/src/main/scala/amf/client/remod/AMFRender.scala]: Looking into this interface i think it will have to at least receive the target vendor, how would conversion be handled? Media type could possibly be added too, or we can just define a default mediatype for each vendor.

Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFRender.scala]: I think that conversation required a dipper analysis (that you are doing) and a discussion based on that analysis.
Using this channel to talk about this could be very complex.

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala (hunk 17 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala:9]: The idea of that argument is to point where the notification was invoked.
Regarding the BaseUnit (the other argument) which information could be use-full? Anyway, that could be a different class which contains the event context, or params.
The kind is just the "name"

File: shared/src/main/scala/amf/client/remod/AMFRender.scala (hunk 19 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFRender.scala]: Added medita type as parameter (following the vendor idea)
The idea of the methods that serialized only on base unit was to serialize the to the origin vendor, or graph if is not present.

**Review by AgustinBettati — DISMISSED** [review]

File: shared/src/main/scala/amf/client/remod/AMFClient.scala [hunk 20]
```diff
@@ -11,29 +11,29 @@ import amf.core.resolution.pipelines.ResolutionPipeline
...
   // how do we return the ErrorHandler that is created by the provider?
   // relation between EH vs BU? should the BU have an EH inside?
   // should parse always return error handler + base unit? (Amf result)
 
   // sync or async?
 
   // content type format, pendiente
-  def parse(url: String, vendor: Option[Vendor] = None): Future[AmfResult] = AMFParser.parse(url, env)
-    // build parsing context?
+  def parse(url: String, vendor: Option[Vendor] = None): Future[AMFResult] = AMFParser.parse(url, configuration)
```
Comment (looseale) [inline shared/src/main/scala/amf/client/remod/AMFClient.scala]: vendor variable is used? The called AMFParser.parse() method didn't receive it, but there is one

File: shared/src/main/scala/amf/client/remod/AMFClient.scala [hunk 21]
```diff
@@ -11,29 +11,29 @@ import amf.core.resolution.pipelines.ResolutionPipeline
...
+  // build parsing context?
 
-  def resolve(bu: BaseUnit): AmfResult = AMFResolver.resolve(bu, env) // clone? BaseUnit.resolved
+  def resolve(bu: BaseUnit): AMFResult = AMFResolver.resolve(bu, configuration) // clone? BaseUnit.resolved
 
-  def render(bu:BaseUnit, target:Vendor):String = AMFRender.render(bu, env)
+  def render(bu: BaseUnit, target: Vendor): String = AMFRenderer.render(bu, configuration)
```
Comment (looseale) [inline shared/src/main/scala/amf/client/remod/AMFClient.scala]: Same here with target (Vendor). The AMFRenderer method didn't receive it, but there is one that receive it as a String... I think it should be typed as Vendor

File: shared/src/main/scala/amf/client/remod/AMFRenderer.scala [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+  def buildAST(bu: BaseUnit, env: AMFConfiguration): YDocument = ???
+
+  /**
+    *
+    * @param bu
+    * @param target media type which specifies a vendor, and optionally a syntax.
+    * @param env
+    * @return
+    */
+  def render(bu: BaseUnit, target: String, env: AMFConfiguration): String = ???
```
Comment (looseale) [inline shared/src/main/scala/amf/client/remod/AMFRenderer.scala]: target should be typed in Vendor for clarification

File: shared/src/main/scala/amf/client/remod/AMFRenderer.scala [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+
+  def buildAST(bu: BaseUnit, target: String, env: AMFConfiguration): YDocument = ???
```
Comment (looseale) [inline shared/src/main/scala/amf/client/remod/AMFRenderer.scala]: Same here

File: shared/src/main/scala/amf/client/remod/AMFResolver.scala [hunk 24]
```diff
@@ -5,8 +5,8 @@ import amf.core.model.document.BaseUnit
 
 private[remod] object AMFResolver {
```
Comment (looseale) [inline shared/src/main/scala/amf/client/remod/AMFResolver.scala]: The name could be AMFTransformer to conform with the internal name change. Also it avoids confusion with AMFResolvers that is the class that contains the ResourceLoader and other things.

File: shared/src/main/scala/amf/client/remod/AMFRenderer.scala (hunk 22 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFRenderer.scala]: The idea is to handled vendors using media type standard, so the users don't have to know our static objects.
We should be able to parse the media type expression.

Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFRenderer.scala]: https://www.mulesoft.org/jira/browse/APIMF-3006

File: shared/src/main/scala/amf/client/remod/AMFClient.scala [hunk 25]
```diff
@@ -11,29 +11,31 @@ import amf.core.resolution.pipelines.ResolutionPipeline
...
 
   // sync or async?
 
   // content type format, pendiente
-  def parse(url: String, vendor: Option[Vendor] = None): Future[AmfResult] = AMFParser.parse(url, env)
-    // build parsing context?
+  def parse(url: String): Future[AMFResult] = AMFParser.parse(url, configuration)
+  // build parsing context?
+
+  def resolve(bu: BaseUnit): AMFResult = AMFTransformer.transform(bu, configuration) // clone? BaseUnit.resolved
```
Comment (nschejtman) [inline shared/src/main/scala/amf/client/remod/AMFClient.scala]: Still calling this resolution?

File: shared/src/main/scala/amf/client/remod/AMFRenderer.scala (hunk 22 above)
Comment (nschejtman) [reply shared/src/main/scala/amf/client/remod/AMFRenderer.scala]: Would rename this to media type then

File: shared/src/main/scala/amf/client/remod/AMFTransformer.scala [hunk 26]
```diff
@@ -0,0 +1,12 @@
+package amf.client.remod
+
+import amf.client.remod.amfcore.resolution.{AMFResolutionPipeline, PipelineName}
+import amf.core.model.document.BaseUnit
+
+private[remod] object AMFTransformer {
+
+  def transform(bu: BaseUnit, env: AMFConfiguration): AMFResult = ???
```
Comment (nschejtman) [inline shared/src/main/scala/amf/client/remod/AMFTransformer.scala]: rename env to congf

File: shared/src/main/scala/amf/core/AMFSerializer.scala [hunk 27]
```diff
@@ -34,7 +34,7 @@ import org.yaml.model.YDocument
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class AMFSerializer(unit: BaseUnit, mediaType: String, vendor: String, env: RenderEnvironment) {
+class AMFSerializer(unit: BaseUnit, mediaType: String, vendor: String, env: RenderConfiguration) {
```
Comment (nschejtman) [inline shared/src/main/scala/amf/core/AMFSerializer.scala:37]: Would make this take an AMFConfiguration for consistency with the AMFCompiler

File: shared/src/main/scala/amf/client/remod/AMFRenderer.scala (hunk 22 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFRenderer.scala]: Change "target" to "mediaType"? I think could be confusing. The argument is the target to which you want to serialize, media type is the argument "type" or "format".
Adding documentation to the argument that needs to be a valid media type should be enough.
(Pending task)

File: shared/src/main/scala/amf/core/AMFSerializer.scala (hunk 27 above)
Comment (pope1838) [reply shared/src/main/scala/amf/core/AMFSerializer.scala:37]: AMFCompiler is still receiving a compiler context.
Changing arguments for this clases is handled at: https://www.mulesoft.org/jira/browse/APIMF-2983

File: shared/src/main/scala/amf/client/remod/AMFResolver.scala (hunk 12 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFResolver.scala]: At the end, I changed as the class name also needed to be updated, so I took the opportunity.

File: shared/src/main/scala/amf/core/services/RuntimeCompiler.scala (hunk 16 above)
Comment (pope1838) [reply shared/src/main/scala/amf/core/services/RuntimeCompiler.scala:65]: But, the formatting now is ok, right?

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala (hunk 17 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/amfcore/config/AMFEventListener.scala:9]: Is ok for you if we close this conversation and discuss the arguments for even listener in the story dedicated to analyze the places where invoke it?
(https://www.mulesoft.org/jira/browse/APIMF-2885)

File: shared/src/main/scala/amf/client/remod/AMFClient.scala (hunk 21 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFClient.scala]: We are having this discussion on the renderer class. Closing this conversation to centralize the subject in just one place.

**Review by looseale — DISMISSED** [review]

**Review by nschejtman — DISMISSED** [review]

**Review by tomsfernandez — DISMISSED** [review]

**Review by looseale — DISMISSED** [review]

**Review by nschejtman — DISMISSED** [review]

File: shared/src/main/scala/amf/client/remod/AMFEnvironment.scala [hunk 28]
```diff
@@ -1,92 +0,0 @@
-package amf.client.remod
```
Comment (AgustinBettati) [inline shared/src/main/scala/amf/client/remod/AMFEnvironment.scala:1]: All of the content has been removed but file was not deleted

**Review by AgustinBettati — APPROVED** [review]
